src/org/traccar/processing/CopyAttributesHandler.java

```diff
@@ -31,21 +32,29 @@ private Position getLastPosition(long deviceId) {
 
     @Override
     protected Position handlePosition(Position position) {
-        String attributesString = Context.getDeviceManager().lookupAttributeString(
-                position.getDeviceId(), "processing.copyAttributes", "", true);
         Position last = getLastPosition(position.getDeviceId());
-        if (attributesString.isEmpty()) {
-            attributesString = Position.KEY_DRIVER_UNIQUE_ID;
-        } else {
-            attributesString += "," + Position.KEY_DRIVER_UNIQUE_ID;
-        }
+
         if (last != null) {
+            String attributesString = Context.getDeviceManager().lookupAttributeString(
+                    position.getDeviceId(), "processing.copyAttributes", null, true);
+
+            if (attributesString == null) {
+                attributesString = Position.KEY_DRIVER_UNIQUE_ID;
+            } else {
+                if (attributesString.isEmpty()) {
```

Comment:
Are we copying all attributes when the parameter is empty? Doesn't make much sense to me at the moment. Did we decide to do it like this for some reason?

Reply:
Also, we don't need attribute string if we want to copy everything.

Reply:
I would make sense to use "null" value to indicate copying of all attributes.